test/services/TxSender.test.ts

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable no-empty */
+import { Wallet, providers } from 'ethers';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { stubConstructor, StubbedInstance } from 'ts-sinon';
+
+import { TxSender } from '../../src/services/TxSender';
+import { GasEstimator } from '../../src/services/GasEstimator';
+import { GasEstimation } from '../../src/types/GasEstimation';
+import { TransactionReceipt, TransactionResponse } from '@ethersproject/abstract-provider';
+import { Wallet as FakeWallet } from '../mocks/Wallet';
+
+const makeFakeTransactionFunction = (result?: { status: number }) => {
+  return async () =>
+    ({
+      wait: () => Promise.resolve(result as TransactionReceipt),
+    } as TransactionResponse);
+};
+
+describe('TxSender', () => {
+  describe('when waitForBlockTime is not provided', () => {
+    it('sets 500ms as default', () => {
+      const subject = new TxSender({ wallet: {} as Wallet, chainId: 'eth' });
+      expect(subject.waitForBlockTime).to.be.equal(500);
+    });
+  });
+
+  describe('#apply', () => {
+    let fakeWallet: Wallet;
+    let fakeProvider: StubbedInstance<providers.BaseProvider>;
+    let txSender: TxSender;
+
+    before(() => {
+      fakeProvider = stubConstructor(providers.JsonRpcProvider);
+      fakeWallet = Wallet.createRandom().connect(fakeProvider);
+    });
+
+    beforeEach(() => {
+      // TxSender needs a new block number every time
+      for (let i = 0; i < 10; i++) {
+        fakeProvider.getBlockNumber.onCall(i /* calls start at 0 */).resolves(100_000 + i);
```

Comment:
this is sync method?

Reply:
There's no need to await here because we're only mocking the resolve value.